client/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+#![no_std]
+
+use core::fmt::Debug;
+use core::marker::PhantomData;
+use gstd::{msg::MessageFuture, prelude::*, ActorId, MessageId};
+use parity_scale_codec::{Decode, Encode};
+
+#[non_exhaustive]
+#[derive(Default)]
+pub struct SendArgs {
+    value: u128,
+    _sender_id: [u8; 32],
+}
+
+#[non_exhaustive]
+pub struct SendTicket<R: Decode + Debug> {
+    // message we are waiting on
+    f: MessageFuture,
+    // silence the compiler
+    _marker: PhantomData<R>,
+}
+
+impl<R: Decode + Debug> SendTicket<R> {
+    pub async fn result(self) -> Result<R, gstd::errors::Error> {
+        let payload = self.f.await?;
+
+        R::decode(&mut payload.as_ref()).map_err(gstd::errors::Error::Decode)
+    }
+
+    pub fn message_id(&self) -> MessageId {
+        self.f.waiting_reply_to
+    }
+}
+
+/// Sender that runs message against gtest::Program
+#[derive(Default)]
```

Comment:
What's the point in this default?

Reply:
Clippy made me do it! https://rust-lang.github.io/rust-clippy/master/#/new_without_default